cpmpy/expressions/core.py

```diff
@@ -554,38 +553,31 @@ def get_bounds(self):
             lb2, ub2 = get_bounds(self.args[1])
             if lb2 <= 0 <= ub2:
                 raise ZeroDivisionError("% by domain containing 0 is not supported")
-            if ub1 <= 0 and ub2 < 0:
-                return lb1, 0
-            elif ub1 <= 0 and lb2 > 0:
-                return 0, ub2
-            elif lb1 >=0 and lb2 > 0:
-                return 0, ub1
             elif ub2 < 0:
-                #lb1 < 0 and ub1 > 0
-                return lb2, 0
+                return lb2 + 1, 0
             elif lb2 > 0:
-                return 0, ub2
+                return 0, ub2 - 1
         elif self.name == 'pow':
             lb1, ub1 = get_bounds(self.args[0])
             lb2, ub2 = get_bounds(self.args[1])
             if lb2 < 0:
                 raise NotImplementedError("Power operator: For integer values, exponent must be non-negative")
             bounds = [lb1**lb2, lb1**ub2, ub1**lb2, ub1**ub2]
-            if ub2 > 0:  # even/uneven behave differently when base is negative
-                bounds += [lb1 ** (ub2 - 1), ub1 ** (ub2 - 1)]
+            if lb1 < 0 and 0 < ub2:  # even/uneven behave differently when base is negative
```

Comment:
Just making sure I get this case: so we add a special case as our get_bounds() functions is not guaranteed to get the tightest bounds? So the actual bounds can be odd while calculated bounds are even and therefore we need this case?

Reply:
No, the problem is that, e.g., (-2)^2 is positive, but (-2)^1 is negative. So the lower bound depends on either the largest or the second largest exponent value when the base term can be negative. So we need to add it.